libyul/optimiser/Suite.cpp

```diff
@@ -86,6 +86,47 @@
 using namespace std;
 using namespace solidity;
 using namespace solidity::yul;
+#ifdef PROFILE_OPTIMIZER_SEQUENCE
+using namespace std::chrono;
+#endif
+
+namespace
+{
+
+#ifdef PROFILE_OPTIMIZER_SEQUENCE
+void outputPerformanceMetrics(map<string, int64_t> const& _metrics)
+{
+	vector<pair<string, int64_t>> durations(_metrics.begin(), _metrics.end());
+	sort(
+		durations.begin(),
+		durations.end(),
+		[](pair<string, int64_t> const& _lhs, pair<string, int64_t> const& _rhs) -> bool
+		{
+			return _lhs.second < _rhs.second;
+		}
+	);
+
+	int64_t totalDurationInMicroseconds = 0;
+	for (auto&& [step, durationInMicroseconds]: durations)
+		totalDurationInMicroseconds += durationInMicroseconds;
+
+	cerr << "Performance metrics of optimizer steps" << endl;
+	cerr << "======================================" << endl;
+	constexpr int64_t microsecondsInSecond = 1000000;
```

Comment:
Wouldn't `double` be better? Maybe we don't even need a cast below in that case?

Reply:
Ahh yes, indeed; in fact I defined it as `double` original, but then wanted to use it for the 1 second check later which would have required an int (long), and I just forgot to move it back.